connectors/slack/template/index.js

```diff
@@ -1,24 +1,295 @@
+  Sdk.debug('New event signature verified');
 
-/**
-* @param ctx {FusebitContext}
-*/
-module.exports = Sdk.createSettingsManager(configure);
+  return ctx.query.dispatch !== undefined ? dispatch(ctx, cb) : respondToSlack(ctx, cb);
+}
+
+function respondToSlack(ctx, cb) {
+  let slackTimestamp = +ctx.headers['x-slack-request-timestamp'] * 1000;
+  Sdk.debug('Slack request time', slackTimestamp);
+
+  // URL Verification Challenge
+  if (ctx.body && ctx.body.type && ctx.body.type === 'url_verification') {
+    Sdk.debug('URL Verification Challenge');
+    cb(null, { body: { challenge: ctx.body.challenge } });
+    return;
+  }
+
+  // Call back to itself to dispatch, wait 100ms, and confirm success to slack
+  // The async dispatch can run beyond the Slack's 3s limit
+
+  Superagent.post(ctx.baseUrl + '/events/ingest?dispatch')
+    .set(ctx.headers)
+    .send(ctx.body)
+    .then(function (result) {
+      Sdk.debug('Dispatch request to self completed with success');
+    })
+    .catch(function (error) {
+      // We silently eat errors here, have to look at realtime logs
+      Sdk.debug('Dispatch request to self completed with error', error);
+    });
+
+  setTimeout(() => {
+    cb(null, { status: 200 });
+  }, 100);
+}
+
+function dispatch(ctx, cb) {
+  let body = ctx.body;
+
+  // Route all other events
+  if (body && body.api_app_id && body.team_id) {
+    storage.ensureStorage(ctx);
+    storage.ensureCache().then(() => {
+      Sdk.debug('Event is for Team ID', body.api_app_id, body.team_id);
+      let handler = storage.teamToHandler[`${body.api_app_id}-${body.team_id}`];
+      if (!handler) {
+        Sdk.debug('No handler registered for this Team ID, not routing');
+        cb();
+        return;
+      }
+
+      Sdk.debug('Handlers registered for this Team ID', handler);
+
+      const promises = handler.map((h) =>
+        Superagent.post(h)
+          .send(body.event)
+          .ok((r) => {
+            if (r.status >= 300) {
+              Sdk.debug('Handler error', h, r.body);
+            }
+            return true;
+          })
+      );
+
+      Promise.all(promises)
+        .then(function (result) {
+          Sdk.debug(`Routed ${handler.length} requests`);
+        })
+        .catch(function (error) {
+          Sdk.debug('Routing error', error);
+        })
+        .finally(function () {
+          cb();
+          return;
+        });
+    });
+  } else {
+    Sdk.debug('No App ID and Team ID provided with event, not routing');
+    cb();
+  }
+}
+
+function registerCore(ctx, cb, isDeregister) {
+  if (!ctx.headers.authorization) {
+    cb(null, {
+      status: 401,
+      body: { message: 'Missing auth token' },
+    });
+    return;
+  }
+
+  storage.ensureStorage(ctx);
+
+  storage.get().then(() => {
+    const [bearerString, token] = ctx.headers.authorization.split(' ');
+    if (bearerString !== 'Bearer' || !Object.keys(storage.tokenToTeam).includes(token)) {
+      cb(null, {
+        status: 401,
```

Comment:
403

Reply:
The check can be simplified to `storage.tokenToTeam[token]`.

Reply:
Fixed